app/controllers/users_controller.rb

```diff
@@ -4,4 +4,17 @@ class UsersController < ApplicationController
   def index
     @users = User.all
   end
+
+  def destroy
+    @user = set_user
+    authorize! :destroy, @user, :message => "You can't delete yourselves"
+    @user.destroy
+    redirect_to users_path, notice: "User #{@user.email} was successfully destroyed."
+  end
+
+  private
+
+  def set_user
+    User.find(params[:id])
```

Comment:
Consider either assigning the user to variable (preferably a cached one) and introduce a before filter. Or even better: aliast this method as `user` and operate on it directly in lines 10-12